### DISCUSSION TITLE
I have a flask application and somewhere it is having a memory leak which is causing disk constraint after a time.

### DISCUSSION THREAD

**Discussion by Mr-Dj-Python** [bloomberg/memray#389] [Q&A]
I have multiple doubts like 

How it can track all the files to find in which specific file memory leak is causing?
How can application API will work?

**Comment by godlygeek** [comment]
By "disk constraint", do you mean that your program is filling one of your drives? If so, I don't think Memray will be very helpful to you. That symptom sounds unlikely to be caused by a memory leak to me. Excess memory usage doesn't turn into disk usage, except as swap - and swap files have a fixed size, and aren't able to grow indefinitely, so they shouldn't be able to fill your disk.

You can see what files are open by a particular process using the `lsof` command, or by looking in `/proc/<the_pid>/fd` where "<the_pid>" is replaced by the process id of the process whose open files you want to see. You can also use a tool like [ncdu](https://dev.yorhel.nl/ncdu) to interactively explore what files/directories on a given disk are taking up the most space.
